src/MooLite/core/inventory/Inventory.ts

```diff
@@ -87,23 +94,65 @@ export class Inventory {
         });
     }
 
-    updateCharacterDrink(actionTypeDrinkSlotsMap: Record<ActionTypeHrid, CharacterConsumable[]>) {
-        this._actionTypeDrinkSlotsMap = actionTypeDrinkSlotsMap;
+    updateCharacterDrink(actionTypeDrinkSlotsMap: Record<ActionTypeHrid, (CharacterConsumable | null)[]>) {
+        for (const actionType in actionTypeDrinkSlotsMap) {
+            const consumables = actionTypeDrinkSlotsMap[actionType];
+
+            if (this._characterDrinks[actionType] == null) {
+                this._characterDrinks[actionType] = consumables;
+            }
+
+            this._characterDrinks[actionType].forEach((storedConsumable, i) => {
+                if (storedConsumable === null) return;
```

Comment:
Nitpick: Can you wrap the return statement here and above in `{}`? Not a fan of the braceless statements